upload.py

```diff
@@ -2709,26 +2721,30 @@ def RealMain(argv, data=None):
   if not response_body.startswith("Issue created.") and \
   not response_body.startswith("Issue updated."):
     sys.exit(0)
-  issue = msg[msg.rfind("/")+1:]
+  issue_id = msg[msg.rfind("/")+1:]
 
   if not uploaded_diff_file:
-    result = UploadSeparatePatches(issue, rpc_server, patchset, data, options)
+    result = UploadSeparatePatches(issue_id, rpc_server, patchset, data, options)
     if not options.download_base:
       patches = result
 
   if not options.download_base:
-    vcs.UploadBaseFiles(issue, rpc_server, patches, patchset, options, files)
+    vcs.UploadBaseFiles(issue_id, rpc_server, patches, patchset, options, files)
 
   payload = {}  # payload for final request
   if options.send_mail:
     payload["send_mail"] = "yes"
     if options.send_patch:
       payload["attach_patch"] = "yes"
-  if options.issue and message:
+  if issue and message:
     payload["message"] = message
   payload = urllib.urlencode(payload)
-  rpc_server.Send("/" + issue + "/upload_complete/" + (patchset or ""),
+  rpc_server.Send("/" + issue_id + "/upload_complete/" + (patchset or ""),
                   payload=payload)
+  if options.issue and not issue:
```

Comment:
Maybe I'll be missing something here, but doesn't this always create a file even when an issue number is given on the command line?

Reply:
Normally if an issue number is given, issue will be filled so the test will fail.

Reply:
ah, right!